config/database.yml

```diff
@@ -11,6 +11,10 @@ default: &default
   connect_timeout: 5
   variables:
     statement_timeout: 5000 # ms
+  sslmode: <%= ENV.fetch('DATABASE_SSL_MODE', ENV.fetch('DATABASE_SSL_CA', nil) ? 'verify-full' : 'disable') %>
```

Comment:
SSL mode can be set in the connection URL already (`DATABASE_URL`).

Reply:
No, I didn't know you could send those params through the URI. I tried and it works. So the whole env vars system was not really needed 🙃. However, I think vars are still useful because they make it easier for the operator.

I case the URI contradicts the env vars, the URI prevails.

Reply:
Why not let users update their `DATABASE_URL` as they please? What is the help of operator? Do we make it actually easier for users with variables?

Reply:
Because if they add a URI param like `?sslca=path/to/cert.pem` then they must mount a volume for `path/to/cert.pem` to be available from inside the container. So they need the operator: https://issues.redhat.com/browse/THREESCALE-11156

Reply:
Looking at this issue, this TLS configuration is it db specific or is it supposed to also apply to other components like redis, generic CA for any component and probably more?

Reply:
this is only for DBs, we have another one for Redis:

https://issues.redhat.com/browse/THREESCALE-11020

Reply:
No need to unset the vars. The operator will make new yaml keywords available, like `dbCaCertRef`. If exists, it'll look for the secret, mount it at the container and set the env variable. If it doesn't exist, then it just won't. Since any change in the yaml triggers a rollout, the new deployment simply won't have the vars set.

Reply:
cool

Reply:
btw I hope we have a better name than `dbCaCertRef` 

Or why not put these values into the existinf system database secret? Why introduce another secret?

It's not part of this PR though. Just saying.

Reply:
Because TLS has it's own `type` of `Secret` in kubernetes: https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets

Reply:
> The TLS Secret type is provided only for convenience. You can create an Opaque type for credentials used for TLS authentication. However, using the defined and public Secret type (kubernetes.io/tls) helps ensure the consistency of Secret format in your project. The API server verifies if the required keys are set for a Secret of this type.

Sounds good in general. But as a user I'd highly prefer my database configuration to be in one place. Especially given that we will have more key basically so the format will ot be fully standard. Or will it?

This is not really in our project though. I guess I can give a suggestion in the operator PR if I ever spot it :)

Reply:
> Especially given that we will have more key basically so the format will not be fully standard. Or will it?

I don't understand the question

> This is not really in our project though. I guess I can give a suggestion in the operator PR if I ever spot it :)

No need to wait for the PR, you can write on the issue: https://issues.redhat.com/browse/THREESCALE-11156

Reply:
We have 3 keys:

- database CA
- ssl cert
- ssl key

while the k8s secret type has 2 standard keys - cert and key. So we still have custom items in there.

Reply:
As I understood it, the CA would require one secret, and another secret for user cert + key.

Reply:
But then for the CA there will be no key. And having more than one secret - even worse. Doesn't depend on us though. I'll just add a comment.

Reply:
The client doesn't need to have the CA key,it only needs the CA certificate to specify from which CA it will accept server certificates.

Reply:
Exactly what I mean. For the CA, there is no key. So you can't have a valid TLS secret.